qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -887,7 +888,15 @@ def get_collections(self,
         collection_class = get_collection_class(collection)
         query = format_query(collection_class, lname=name, collection=collection, id=col_id)
 
-        # try:
+        if not heavy and projection is None:
+            projection = collection_class.col()
+            for item in [
+                    "contributed_values", "records", "records_obj", "update_relations", "lname", "extra",
```

Comment:
`lname` is already excluded in all queries.

This doesn't look general enough, it seems customized to one use case that doesn't need extra nor records nor contributed_values. Also, and most importantly, records are specific to Datasets not all kinds of collections.
I'll push some code to the PR.